internal/postprocessor/main.go

```diff
@@ -268,3 +243,163 @@ func docURL(cloudDir, importPath string) (string, error) {
 	pkgPath := strings.TrimPrefix(strings.TrimPrefix(importPath, mod), "/")
 	return "https://cloud.google.com/go/docs/reference/" + mod + "/latest/" + pkgPath, nil
 }
+
+func (c *config) AmendPRDescription(ctx context.Context) error {
+	log.Println("Amending PR title and body")
+	pr, err := c.getPR(ctx)
+	if err != nil {
+		return err
+	}
+	newPRTitle, newPRBody, err := c.processCommit(*pr.Title, *pr.Body)
+	if err != nil {
+		return err
+	}
+	return c.writePRCommitToFile(newPRTitle, newPRBody)
+}
+
+func (c *config) processCommit(title, body string) (string, string, error) {
+	var newPRTitle string
+	var commitTitle string
+	var commitTitleIndex int
+
+	bodySlice := strings.Split(body, "\n")
+	for index, line := range bodySlice {
+		if strings.Contains(line, apiNameOwlBotScope) {
+			commitTitle = line
+			commitTitleIndex = index
+			continue
+		}
+		// When OwlBot generates the commit body, after commit titles it provides 'Source-Link's.
+		// The source-link pointing to the googleapis/googleapis repo commit allows us to extract 
+		// hash and find files changed in order to identify the commit's scope.
+		if !strings.Contains(line, "googleapis/googleapis/") {
+			continue
+		}
+		hash := extractHashFromLine(line)
+		scope, err := c.getScopeFromGoogleapisCommitHash(hash)
+		if err != nil {
+			return "", "", err
+		}
+		if newPRTitle == "" {
+			newPRTitle = updateCommitTitle(title, scope)
+			continue
+		}
+		newCommitTitle := updateCommitTitle(commitTitle, scope)
+		bodySlice[commitTitleIndex] = newCommitTitle
+	}
+	body = strings.Join(bodySlice, "\n")
+	return newPRTitle, body, nil
+}
+
+func (c *config) getPR(ctx context.Context) (*github.PullRequest, error) {
+	client := github.NewClient(nil)
+	prs, _, err := client.PullRequests.List(ctx, c.githubUsername, "google-cloud-go", nil)
+	if err != nil {
+		return nil, err
+	}
+	pr := c.findValidPR(ctx, prs)
+	if pr == nil {
+		return nil, errors.New("no PR found")
+	}
+	return pr, nil
+}
+
+func (c *config) findValidPR(ctx context.Context, prs []*github.PullRequest) *github.PullRequest {
```

Comment:
I just meant that you can get rid of this whole function and move it into the other.

Reply:
Done.